src/web_interface/components/miscellaneous_routes.py

```diff
@@ -68,5 +70,39 @@ def _app_delete_firmware(self, uid):
         if not is_firmware:
             return render_template('error.html', message='Firmware not found in database: {}'.format(uid))
         with ConnectTo(AdminDbInterface, config=self._config) as sc:
-            deleted_virtual_file_path_entries, deleted_files = sc.delete_firmware(uid)
-        return render_template('delete_firmware.html', deleted_vps=deleted_virtual_file_path_entries, deleted_files=deleted_files, uid=uid)
+            deleted_virtual_path_entries, deleted_files = sc.delete_firmware(uid)
+        return render_template('delete_firmware.html', deleted_vps=deleted_virtual_path_entries, deleted_files=deleted_files, uid=uid)
+
+    @roles_accepted(*PRIVILEGES['delete'])
+    def _app_find_missing_analyses(self):
+        data = self._find_missing_files()
+        data.update(self._find_missing_analyses())
+        return render_template('find_missing_analyses.html', data=data)
+
+    def _find_missing_files(self):
+        start = time()
+        with ConnectTo(FrontEndDbInterface, config=self._config) as db:
+            parent_to_included = db.find_missing_files()
+        return {
+            'missing_files': list(parent_to_included.items()),
+            'missing_files_count': self._count_values(parent_to_included),
+            'missing_files_duration': self._get_duration(start),
+        }
+
+    def _find_missing_analyses(self):
+        start = time()
+        with ConnectTo(FrontEndDbInterface, config=self._config) as db:
+            missing_analyses = db.find_missing_analyses()
+        return {
+            'missing_analyses': missing_analyses,
+            'missing_analyses_count': self._count_values(missing_analyses),
+            'missing_analyses_duration': self._get_duration(start),
+        }
+
+    @staticmethod
+    def _count_values(dictionary: Dict[str, Sized]) -> int:
+        return sum(len(e) for e in dictionary.values())
+
+    @staticmethod
+    def _get_duration(start: float) -> str:
+        return '{}s'.format(si_format(time() - start, precision=2))
```

Comment:
Ah, do you really need an external library for this? That looks like an easy arithmetic operation

Reply:
Do you really think we should write our own library for formatting SI units if we could use an existing one from PyPI just as well?

Reply:
Generally sure, take the existing code. But we're not talking writing a library here, are we? Isn't this just division by `10**x` for some static `x`. Maybe I'm at a miss here.